Assets/Scripts/Battlefield/CreatureScripts/CharacterFactory.cs

```diff
@@ -21,15 +23,12 @@ public class CharacterFactory : MonoBehaviour
         public Transform unitHolder;
     
         void Awake()
-        {
+        {             
+            IEnumerable<IUnit> unitIDs = (SceneSharing.squadID != -1) ? new Squad(SceneSharing.squadID).Units : GetAllUnits();
```

Comment:
Just make it a list rather than IEnumerable

Reply:
Has to be an IEnumerable to use yield return

Reply:
Doesn't need to be if you use alternate method in Units

Reply:
Resolved in 40c6f71